src/wallet/scriptpubkeyman.cpp

```diff
@@ -594,3 +594,54 @@ bool LegacyScriptPubKeyMan::CanGenerateKeys()
     LOCK(cs_KeyStore);
     return IsHDEnabled() || !CanSupportFeature(FEATURE_HD);
 }
+
+CPubKey LegacyScriptPubKeyMan::GenerateNewSeed()
+{
+    assert(!IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS));
+    CKey key;
+    key.MakeNewKey(true);
+    return DeriveNewSeed(key);
+}
+
+CPubKey LegacyScriptPubKeyMan::DeriveNewSeed(const CKey& key)
```

Comment:
They were originally together but later split apart to allow for the `sethdseed` RPC. I suppose it should be `DeriveAndSetHDSeed` which calls `SetHDSeed`. But since this is moved from how it is currently, I will leave it as is.